presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -131,6 +133,56 @@ public void testAggregation()
                                                         anyTree(tableScan("orders", ImmutableMap.of("totalprice", "totalprice")))))))));
     }
 
+    @Test
+    public void testWindowAfterJoin()
+    {
+        // Window Partition key is a super set of join key.
+        assertDistributedPlan("SELECT rank() OVER (partition by o.orderstatus) FROM orders o JOIN lineitem l ON o.orderstatus = l.linestatus",
```

Comment:
could you add another column to window partition by so that window partition is superset of child partitioning?